services/cd-service/pkg/service/batch.go

```diff
@@ -166,21 +167,67 @@ func (d *BatchServer) processAction(
 			Application:   act.Application,
 			Version:       act.Version,
 			LockBehaviour: b,
-		}, nil
+		}, nil, nil
 	case *api.BatchAction_DeleteEnvFromApp:
 		act := action.DeleteEnvFromApp
 		return &repository.DeleteEnvFromApp{
 			Environment: act.Environment,
 			Application: act.Application,
-		}, nil
+		}, nil, nil
+	case *api.BatchAction_ReleaseTrain:
+		in := action.ReleaseTrain
```

Comment:
i think these checks should happen in the release train transformer.

Reply:
I just moved them around from here: https://github.com/freiheit-com/kuberpult/blob/ebfbe67f7382150b5c6070cd88aecfd7f4c5cbc5/services/cd-service/pkg/service/deploy.go#L88
I generally think it's fine to do such preflight check upfront.

Reply:
fine for now.
I will look if we can find a general approach on where to do the validations.